app/controllers/products_controller.rb

```diff
@@ -3,5 +3,7 @@ def index
   end
 
   def show
+    @product = Product.find_by_id(params[:id])
```

Comment:
Thanks for this.
The reason why we used `find_by_id` is that `find` raises an exception when a record matching that *id* is not found. However, `find_by_id` returns nil in such scenario.